pkg/gateway/lb/handler/grpc/handler.go

```diff
@@ -493,6 +494,46 @@ func (s *server) search(ctx context.Context, cfg *payload.Search_Config,
 			if num != 0 && len(res.GetResults()) > num {
 				res.Results = res.GetResults()[:num]
 			}
+
+			if errors.Is(ectx.Err(), context.DeadlineExceeded) {
+				if len(res.GetResults()) == 0 {
+					err = status.WrapWithDeadlineExceeded(
+						"error search result length is 0",
+						errors.ErrEmptySearchResult,
+						&errdetails.RequestInfo{
+							RequestId:   cfg.GetRequestId(),
+							ServingData: errdetails.Serialize(cfg),
+						},
+						&errdetails.ResourceInfo{
+							ResourceType: errdetails.ValdGRPCResourceTypePrefix + "/vald.v1.search",
+							ResourceName: fmt.Sprintf("%s: %s(%s) to %v", apiName, s.name, s.ip, s.gateway.Addrs(ctx)),
+						}, info.Get(),
+					)
+					if span != nil {
+						span.SetStatus(trace.StatusCodeDeadlineExceeded(err.Error()))
+					}
+					return nil, err
+				}
+				if min != 0 && len(res.GetResults()) < min {
+					err = status.WrapWithDeadlineExceeded(
+						"error search result length less than minimum number",
```

Comment:
How about showing the real result length?
It is more convenient for the user.

Reply:
Thank you. it looks good to me!
I modified it based on your suggestion. 
Below is an example when a length is 100.

```
error search result length is 100
```
What do you think?

diff: https://github.com/vdaas/vald/pull/1546/commits/30c376e68b4d97ae157911cbb1d402c72c6b8c51